File: python/844.Backspace-String-Compare.py
class Solution:
    def build(self, string):
        stack = []
        for c in string:
            if c == "#":
                if stack:
                    stack.pop()
            else:
                stack.append(c)
        return stack

    def backspaceCompare(self, s: str, t: str) -> bool:
        return self.build(s) == self.build(t)

File: javascript/844.Backspace-String-Compare.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var backspaceCompare = function (s, t) {
  const sStk = [];
  const tStk = [];

  for (const c of s) {
    if (c === "#") {
      sStk.pop();
    } else {
      sStk.push(c);
    }
  }

  for (const c of t) {
    if (c === "#") {
      tStk.pop();
    } else {
      tStk.push(c);
    }
  }

  return sStk.join("") === tStk.join("");
};
